// @dasha.ai/sdk@0.11.2/dist/rest-api/package-permision.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackagePermissionApi = void 0;
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/core/api"));
class PackagePermissionApi extends base.Api {
    constructor(account, log) {
        super(api.PackagePermissionApi, account, log);
    }
    async getMyPermissions(skip, take) {
        const response = await this.api.apiV1PackagesPermissionOwnListGet(skip, take, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getPackagePermissions(name, skip, take) {
        const response = await this.api.apiV1PackagesPermissionListGet(name, skip, take, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getPackagePermissionsById(packageId, skip, take) {
        const response = await this.api.apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageId, skip, take, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getMyPackagePermissionsById(packageId) {
        const response = await this.api.apiV1PackagesPermissionPackageDescriptionIdGet(packageId, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getMyPackagePermissions(name) {
        const response = await this.api.apiV1PackagesPermissionGet(name, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getMyPermissionsForPackage(name) {
        const response = await this.api.apiV1PackagesPermissionGet(name, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async deletePermission(name, customerId) {
        const response = await this.api.apiV1PackagesPermissionDelete({ name: name, customerId: customerId }, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async garantPermission(name, customerId, permission) {
        const response = await this.api.apiV1PackagesPermissionPost({ name: name, customerId: customerId, accessPermissions: permission }, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
}
exports.PackagePermissionApi = PackagePermissionApi;
