// @dasha.ai/sdk@0.11.2/dist/rest-api/prompt.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromptApi = void 0;
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/testsystem/api"));
class PromptApi extends base.Api {
    constructor(account, log) {
        super(api.PromptApi, account, log);
    }
    async addPrompt(request) {
        const response = await this.api.apiV1PromptPost(request, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async deletePrompt(id) {
        const response = await this.api.apiV1PromptIdDelete(id, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async updatePrompt(id, request) {
        const response = await this.api.apiV1PromptIdPatch(id, request, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getPrompt(id) {
        const response = await this.api.apiV1PromptIdGet(id, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async listPrompts(skip, take) {
        const response = await this.api.apiV1PromptGet(skip, take, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async listPromptVersions(promptId, skip, take) {
        const response = await this.api.apiV1PromptPromptIdVersionsGet(promptId, skip, take, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async addPromptVersion(promptId, dto) {
        const response = await this.api.apiV1PromptPromptIdVersionsPost(promptId, dto, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async deletePromptVersion(promptId) {
        const response = await this.api.apiV1PromptVersionsIdDelete(promptId, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async callPrompt(dto, environment, version) {
        const response = await this.api.apiV1PromptCallPost(environment, version, dto, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
}
exports.PromptApi = PromptApi;
