// @dasha.ai/sdk@0.11.2/dist/rest-api/sip-endpoint.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SipEndpointApi = void 0;
const errors = __importStar(require("../errors"));
const log_1 = require("../log");
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/core/api"));
class SipEndpointApi extends base.Api {
    constructor(account, log) {
        super(api.SIPEndpointApi, account, log);
    }
    async deleteConfig(name, options) {
        const getResponse = await this.api.apiV1SipEndpointByNameGet(name, {
            validateStatus: (status) => status < 400 || status === 404,
        });
        if (getResponse.status === 404) {
            if (!(options === null || options === void 0 ? void 0 : options.quiet))
                (0, log_1.logAndThrow)(this.log, new errors.SipConfigNotFoundError(name));
            return;
        }
        await this.api.apiV1SipEndpointIdDelete(getResponse.data.id, {
            validateStatus: (status) => status < 400,
        });
    }
    async createConfig(request) {
        const response = await this.api.apiV1SipEndpointPost(request, {
            validateStatus: (status) => status < 400 || status === 409,
        });
        if (response.status === 409) {
            const errorData = response.data;
            (0, log_1.logAndThrow)(this.log, new errors.SipConfigDuplicateError(errorData.reason));
        }
        return response.data;
    }
    async getConfigByName(name) {
        const response = await this.api.apiV1SipEndpointByNameGet(name, {
            validateStatus: (status) => status < 400 || status === 404,
        });
        if (response.status === 404) {
            (0, log_1.logAndThrow)(this.log, new errors.SipConfigNotFoundError(name));
        }
        return response.data;
    }
    async listConfigs() {
        const response = await this.api.apiV1SipEndpointGet(undefined, // skip
        undefined, // take
        {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async updateConfig(name, request) {
        const response = await this.api.apiV1SipEndpointByNameGet(name, {
            validateStatus: (status) => status < 400 || status === 404,
        });
        if (response.status === 404) {
            (0, log_1.logAndThrow)(this.log, new errors.SipConfigNotFoundError(name));
        }
        const patchResponse = await this.api.apiV1SipEndpointIdPatch(response.data.id, request, {
            validateStatus: (status) => status < 400 || status === 409,
        });
        if (patchResponse.status === 409) {
            const errorData = patchResponse.data;
            (0, log_1.logAndThrow)(this.log, new errors.SipConfigDuplicateError(errorData.reason));
        }
        return patchResponse.data;
    }
    async getInboundCalls(fromDate, toDate, fromStatus, toStatus, skip, take) {
        const response = await this.api.apiV1SipEndpointReportsGet(fromDate, toDate, fromStatus, toStatus, skip, take);
        return response.data;
    }
    async getEndpointStatus(endpointId, errThreshold) {
        try {
            const response = await this.api.apiV1SipEndpointStatusEndpointIdGet(endpointId, errThreshold, {
                validateStatus: (status) => true,
            });
            if (response.status < 300) {
                return undefined;
            }
            return response.data;
        }
        catch (e) {
            return 'API Error';
        }
    }
}
exports.SipEndpointApi = SipEndpointApi;
