// @dasha.ai/sdk@0.11.2/dist/rest-api/usage.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageApi = void 0;
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/core/api"));
class UsageApi extends base.Api {
    constructor(account, log) {
        super(api.UsageApi, account, log);
    }
    async get(take, skip, filters) {
        const response = await this.api.apiV1UsageBillablePost(skip, take, {
            filters: filters
        }, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getForJob(jobId) {
        const response = await this.api.apiV1UsageBillableJobIdGet(jobId, {
            validateStatus: (status) => status < 400 || status == 404,
        });
        if (response.status === 404) {
            return undefined;
        }
        return response.data;
    }
    async getFilters(filters, fillFilters) {
        const response = await this.api.apiV1UsageFiltersPost({
            filters: filters,
            fillFilters: fillFilters
        }, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
}
exports.UsageApi = UsageApi;
