// @dasha.ai/sdk@0.11.2/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.CircularBuffer = exports.TranscriptionParser = exports.QueuedConversation = exports.ConversationQueue = exports.log = exports.deploy = exports.SingleConversation = exports.configure = exports.CancelToken = exports.Application = exports.support = exports.prompt = exports.coreapi = exports.testapi = exports.testsystem = exports.gpt = exports.onlineprofiler = exports.jobs = exports.filters = exports.events = exports.usage = exports.packages = exports.tts = exports.stt = exports.sip = exports.profiler = exports.nlu = exports.groups = exports.errors = exports.diagnostics = exports.customer = exports.multichannelChat = exports.chat = exports.auth = exports.audio = exports.applications = exports.account = void 0;
exports.account = __importStar(require("./account"));
exports.applications = __importStar(require("./applications"));
exports.audio = __importStar(require("./audio"));
exports.auth = __importStar(require("./auth"));
exports.chat = __importStar(require("./chat"));
exports.multichannelChat = __importStar(require("./multichannel-chat"));
exports.customer = __importStar(require("./customer"));
exports.diagnostics = __importStar(require("./diagnostics"));
exports.errors = __importStar(require("./errors"));
exports.groups = __importStar(require("./groups"));
exports.nlu = __importStar(require("./nlu"));
exports.profiler = __importStar(require("./profiler"));
exports.sip = __importStar(require("./sip"));
exports.stt = __importStar(require("./stt"));
exports.tts = __importStar(require("./tts"));
exports.packages = __importStar(require("./package-manager"));
exports.usage = __importStar(require("./usage"));
exports.events = __importStar(require("./events"));
exports.filters = __importStar(require("./filters"));
exports.jobs = __importStar(require("./jobs"));
exports.onlineprofiler = __importStar(require("./online-profiler"));
exports.gpt = __importStar(require("./gpt"));
exports.testsystem = __importStar(require("./testsytem"));
exports.testapi = __importStar(require("./rest-api/generated/testsystem"));
exports.coreapi = __importStar(require("./rest-api/generated/core"));
exports.prompt = __importStar(require("./prompt"));
exports.support = __importStar(require("./support"));
var application_1 = require("./application");
Object.defineProperty(exports, "Application", { enumerable: true, get: function () { return application_1.Application; } });
var cancel_1 = require("./cancel");
Object.defineProperty(exports, "CancelToken", { enumerable: true, get: function () { return cancel_1.CancelToken; } });
var configure_1 = require("./configure");
Object.defineProperty(exports, "configure", { enumerable: true, get: function () { return configure_1.configure; } });
var conversation_1 = require("./conversation");
Object.defineProperty(exports, "SingleConversation", { enumerable: true, get: function () { return conversation_1.SingleConversation; } });
var deploy_1 = require("./deploy");
Object.defineProperty(exports, "deploy", { enumerable: true, get: function () { return deploy_1.deploy; } });
var log_1 = require("./log");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return log_1.log; } });
var queue_1 = require("./queue");
Object.defineProperty(exports, "ConversationQueue", { enumerable: true, get: function () { return queue_1.ConversationQueue; } });
Object.defineProperty(exports, "QueuedConversation", { enumerable: true, get: function () { return queue_1.QueuedConversation; } });
var transcription_1 = require("./transcription");
Object.defineProperty(exports, "TranscriptionParser", { enumerable: true, get: function () { return transcription_1.TranscriptionParser; } });
var circular_buffer_1 = require("./circular-buffer");
Object.defineProperty(exports, "CircularBuffer", { enumerable: true, get: function () { return circular_buffer_1.CircularBuffer; } });
/** @internal Not available for web bundle */
const sdk_observer_1 = require("./sdk-observer");
if (process.env.DASHA_SDK_OBSERVER_PORT !== undefined &&
    process.env.DASHA_SDK_OBSERVER_PORT !== "") {
    const port = process.env.DASHA_SDK_OBSERVER_PORT;
    let cachedConversationCount = parseInt((_a = process.env.DASHA_SDK_OBSERVER_CACHED_CONVERSATION_COUNT) !== null && _a !== void 0 ? _a : "");
    if (Number.isNaN(cachedConversationCount))
        cachedConversationCount = 10000;
    sdk_observer_1.SdkObserver === null || sdk_observer_1.SdkObserver === void 0 ? void 0 : sdk_observer_1.SdkObserver.initialize(port, cachedConversationCount);
}
