// @dasha.ai/sdk@0.11.2/dist/zip.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.directory = directory;
exports.file = file;
exports.jszipFiles = jszipFiles;
exports.application = application;
exports.unzip = unzip;
const log_1 = require("./log");
const zip_helpers_1 = require("./zip-helpers");
const errors = __importStar(require("./errors"));
const jszip_1 = __importDefault(require("jszip"));
const path_1 = __importDefault(require("path"));
// the code here is written as if `archiver` was an optional dependency
// based on the assumption that we may want to work with pre-packed applications
// in the future
/** @internal */
async function directory(path, log) {
    const { default: archiver } = await Promise.resolve().then(() => __importStar(require("archiver"))).catch(() => {
        (0, log_1.logAndThrow)(log, new errors.OptionalDependencyNotFoundError("archiver"));
    });
    return new Promise((resolve, reject) => {
        const zip = archiver("zip");
        const chunks = [];
        zip.on("error", (error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
        zip.on("warning", (error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
        zip.on("data", (chunk) => chunks.push(chunk));
        zip.on("finish", () => resolve(Buffer.concat(chunks)));
        zip.directory(path, false);
        zip.finalize().catch((error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
    });
}
/** @internal */
async function jszipFile(name, content, log) {
    const zip = new jszip_1.default();
    zip.file(name, content);
    const array = await zip.generateAsync({ type: "uint8array" }).catch((error) => {
        (0, log_1.logAndThrow)(log, new errors.FileSystemError(error));
    });
    return array;
}
/** @internal */
async function archiverFile(name, value, log) {
    const { default: archiver } = await Promise.resolve().then(() => __importStar(require("archiver")));
    const stream = await Promise.resolve().then(() => __importStar(require("stream")));
    return new Promise((resolve, reject) => {
        const zip = archiver("zip");
        const chunks = [];
        zip.on("error", (error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
        zip.on("warning", (error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
        zip.on("data", (chunk) => chunks.push(chunk));
        zip.on("finish", () => resolve(Buffer.concat(chunks)));
        zip.append(value, { name });
        zip.finalize().catch((error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
    });
}
/** @internal */
async function file(name, content, log) {
    try {
        await Promise.resolve().then(() => __importStar(require("archiver"))).catch(() => {
            (0, log_1.logAndThrow)(log, new errors.OptionalDependencyNotFoundError("archiver"));
        });
        return await archiverFile(name, content, log);
    }
    catch (e) {
        return await jszipFile(name, content, log);
    }
}
/** @internal */
async function jszipFiles(zipData, log) {
    const zip = new jszip_1.default();
    for (const [path, content] of Object.entries(zipData)) {
        zip.file(path, content);
    }
    return await zip
        .generateAsync({ type: "uint8array" })
        .catch((error) => (0, log_1.logAndThrow)(log, new errors.FileSystemError(error)));
}
/** @internal */
async function archiverFiles(zipData, log) {
    const { default: archiver } = await Promise.resolve().then(() => __importStar(require("archiver")));
    return new Promise((resolve, reject) => {
        const zip = archiver("zip");
        const chunks = [];
        zip.on("error", (error) => reject(error));
        zip.on("warning", (error) => reject(error));
        zip.on("data", (chunk) => chunks.push(chunk));
        zip.on("finish", () => resolve(Buffer.concat(chunks)));
        for (const [name, content] of Object.entries(zipData))
            zip.append(content, { name });
        zip.finalize().catch((error) => reject((0, log_1.logAndReturn)(log, new errors.FileSystemError(error))));
    });
}
/** @internal */
async function application(appFolderPath, log) {
    const zipData = await (0, zip_helpers_1.prepareAppZipData)(appFolderPath);
    try {
        await Promise.resolve().then(() => __importStar(require("archiver"))).catch(() => {
            (0, log_1.logAndThrow)(log, new errors.OptionalDependencyNotFoundError("archiver"));
        });
        return await archiverFiles(zipData, log).catch((e) => (0, log_1.logAndThrow)(log, new errors.FileSystemError(e)));
    }
    catch (e) {
        return await jszipFiles(zipData, log).catch((e) => (0, log_1.logAndThrow)(log, new errors.FileSystemError(e)));
    }
}
/**
 * Attention prepare directory before unzip
 */
async function unzip(directoryPath, buffer, fs) {
    const zip = await jszip_1.default.loadAsync(buffer);
    for (const [pathEntry,] of Object.entries(zip.files).filter(x => x[1].dir)) {
        const resolvedPath = path_1.default.resolve(directoryPath, pathEntry);
        await fs.mkdir(resolvedPath, { recursive: true });
    }
    for (const [pathEntry, entry] of Object.entries(zip.files).filter(x => !x[1].dir)) {
        const resolvedPath = path_1.default.resolve(directoryPath, pathEntry);
        await fs.mkdir(path_1.default.dirname(resolvedPath), { recursive: true });
        await fs.writeFile(resolvedPath, await entry.async("uint8array"));
    }
}
