// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/models/filter-name.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterName = void 0;
/**
 *
 * @export
 * @enum {string}
 */
exports.FilterName = {
    EventTypes: 'EventTypes',
    UsageTypes: 'UsageTypes',
    UsageSubTypes: 'UsageSubTypes',
    DateRange: 'DateRange',
    ApplicationNames: 'ApplicationNames',
    GroupIds: 'GroupIds',
    ApplicationIds: 'ApplicationIds',
    CustomerIds: 'CustomerIds',
    JobStatuses: 'JobStatuses',
    StartedTime: 'StartedTime',
    CompletedTime: 'CompletedTime',
    ChannelTypes: 'ChannelTypes',
    TimeStamps: 'TimeStamps',
    JobIds: 'JobIds',
    Ids: 'Ids'
};
