// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/agents-environment-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentsEnvironmentApi = exports.AgentsEnvironmentApiFactory = exports.AgentsEnvironmentApiFp = exports.AgentsEnvironmentApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * AgentsEnvironmentApi - axios parameter creator
 * @export
 */
const AgentsEnvironmentApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {HiddenStateEnum} [hiddenState]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsGet: async (hiddenState, skip, take, options = {}) => {
            const localVarPath = `/api/v1/agents/environments`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (hiddenState !== undefined) {
                localVarQueryParameter['hiddenState'] = hiddenState;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsIdGet: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsEnvironmentsIdGet', 'id', id);
            const localVarPath = `/api/v1/agents/environments/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {AgentsEnvironmentUpdateDTO} [agentsEnvironmentUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsIdPatch: async (id, agentsEnvironmentUpdateDTO, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsEnvironmentsIdPatch', 'id', id);
            const localVarPath = `/api/v1/agents/environments/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(agentsEnvironmentUpdateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {AgentsEnvironmentCreateDTO} [agentsEnvironmentCreateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsPost: async (agentsEnvironmentCreateDTO, options = {}) => {
            const localVarPath = `/api/v1/agents/environments`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(agentsEnvironmentCreateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.AgentsEnvironmentApiAxiosParamCreator = AgentsEnvironmentApiAxiosParamCreator;
/**
 * AgentsEnvironmentApi - functional programming interface
 * @export
 */
const AgentsEnvironmentApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.AgentsEnvironmentApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {HiddenStateEnum} [hiddenState]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsEnvironmentsGet(hiddenState, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsEnvironmentsGet(hiddenState, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentsEnvironmentApi.apiV1AgentsEnvironmentsGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsEnvironmentsIdGet(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsEnvironmentsIdGet(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentsEnvironmentApi.apiV1AgentsEnvironmentsIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {AgentsEnvironmentUpdateDTO} [agentsEnvironmentUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsEnvironmentsIdPatch(id, agentsEnvironmentUpdateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsEnvironmentsIdPatch(id, agentsEnvironmentUpdateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentsEnvironmentApi.apiV1AgentsEnvironmentsIdPatch']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {AgentsEnvironmentCreateDTO} [agentsEnvironmentCreateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsEnvironmentsPost(agentsEnvironmentCreateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsEnvironmentsPost(agentsEnvironmentCreateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentsEnvironmentApi.apiV1AgentsEnvironmentsPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.AgentsEnvironmentApiFp = AgentsEnvironmentApiFp;
/**
 * AgentsEnvironmentApi - factory interface
 * @export
 */
const AgentsEnvironmentApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.AgentsEnvironmentApiFp)(configuration);
    return {
        /**
         *
         * @param {HiddenStateEnum} [hiddenState]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsGet(hiddenState, skip, take, options) {
            return localVarFp.apiV1AgentsEnvironmentsGet(hiddenState, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsIdGet(id, options) {
            return localVarFp.apiV1AgentsEnvironmentsIdGet(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {AgentsEnvironmentUpdateDTO} [agentsEnvironmentUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsIdPatch(id, agentsEnvironmentUpdateDTO, options) {
            return localVarFp.apiV1AgentsEnvironmentsIdPatch(id, agentsEnvironmentUpdateDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {AgentsEnvironmentCreateDTO} [agentsEnvironmentCreateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsEnvironmentsPost(agentsEnvironmentCreateDTO, options) {
            return localVarFp.apiV1AgentsEnvironmentsPost(agentsEnvironmentCreateDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.AgentsEnvironmentApiFactory = AgentsEnvironmentApiFactory;
/**
 * AgentsEnvironmentApi - object-oriented interface
 * @export
 * @class AgentsEnvironmentApi
 * @extends {BaseAPI}
 */
class AgentsEnvironmentApi extends base_1.BaseAPI {
    /**
     *
     * @param {HiddenStateEnum} [hiddenState]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentsEnvironmentApi
     */
    apiV1AgentsEnvironmentsGet(hiddenState, skip, take, options) {
        return (0, exports.AgentsEnvironmentApiFp)(this.configuration).apiV1AgentsEnvironmentsGet(hiddenState, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentsEnvironmentApi
     */
    apiV1AgentsEnvironmentsIdGet(id, options) {
        return (0, exports.AgentsEnvironmentApiFp)(this.configuration).apiV1AgentsEnvironmentsIdGet(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {AgentsEnvironmentUpdateDTO} [agentsEnvironmentUpdateDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentsEnvironmentApi
     */
    apiV1AgentsEnvironmentsIdPatch(id, agentsEnvironmentUpdateDTO, options) {
        return (0, exports.AgentsEnvironmentApiFp)(this.configuration).apiV1AgentsEnvironmentsIdPatch(id, agentsEnvironmentUpdateDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {AgentsEnvironmentCreateDTO} [agentsEnvironmentCreateDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentsEnvironmentApi
     */
    apiV1AgentsEnvironmentsPost(agentsEnvironmentCreateDTO, options) {
        return (0, exports.AgentsEnvironmentApiFp)(this.configuration).apiV1AgentsEnvironmentsPost(agentsEnvironmentCreateDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.AgentsEnvironmentApi = AgentsEnvironmentApi;
