// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/agent-version-in-environment-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentVersionInEnvironmentApi = exports.AgentVersionInEnvironmentApiFactory = exports.AgentVersionInEnvironmentApiFp = exports.AgentVersionInEnvironmentApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * AgentVersionInEnvironmentApi - axios parameter creator
 * @export
 */
const AgentVersionInEnvironmentApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} [agentId]
         * @param {string} [environmentId]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksGet: async (agentId, environmentId, skip, take, options = {}) => {
            const localVarPath = `/api/v1/agents/links`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (agentId !== undefined) {
                localVarQueryParameter['agentId'] = agentId;
            }
            if (environmentId !== undefined) {
                localVarQueryParameter['environmentId'] = environmentId;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksIdDelete: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsLinksIdDelete', 'id', id);
            const localVarPath = `/api/v1/agents/links/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksIdGet: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsLinksIdGet', 'id', id);
            const localVarPath = `/api/v1/agents/links/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {AgentVersionInEnvironmentUpdateDTO} [agentVersionInEnvironmentUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksIdPatch: async (id, agentVersionInEnvironmentUpdateDTO, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsLinksIdPatch', 'id', id);
            const localVarPath = `/api/v1/agents/links/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(agentVersionInEnvironmentUpdateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {AgentVersionInEnvironmentCreateDTO} [agentVersionInEnvironmentCreateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksPost: async (agentVersionInEnvironmentCreateDTO, options = {}) => {
            const localVarPath = `/api/v1/agents/links`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(agentVersionInEnvironmentCreateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.AgentVersionInEnvironmentApiAxiosParamCreator = AgentVersionInEnvironmentApiAxiosParamCreator;
/**
 * AgentVersionInEnvironmentApi - functional programming interface
 * @export
 */
const AgentVersionInEnvironmentApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.AgentVersionInEnvironmentApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} [agentId]
         * @param {string} [environmentId]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsLinksGet(agentId, environmentId, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsLinksGet(agentId, environmentId, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionInEnvironmentApi.apiV1AgentsLinksGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsLinksIdDelete(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsLinksIdDelete(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionInEnvironmentApi.apiV1AgentsLinksIdDelete']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsLinksIdGet(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsLinksIdGet(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionInEnvironmentApi.apiV1AgentsLinksIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {AgentVersionInEnvironmentUpdateDTO} [agentVersionInEnvironmentUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsLinksIdPatch(id, agentVersionInEnvironmentUpdateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsLinksIdPatch(id, agentVersionInEnvironmentUpdateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionInEnvironmentApi.apiV1AgentsLinksIdPatch']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {AgentVersionInEnvironmentCreateDTO} [agentVersionInEnvironmentCreateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsLinksPost(agentVersionInEnvironmentCreateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsLinksPost(agentVersionInEnvironmentCreateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionInEnvironmentApi.apiV1AgentsLinksPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.AgentVersionInEnvironmentApiFp = AgentVersionInEnvironmentApiFp;
/**
 * AgentVersionInEnvironmentApi - factory interface
 * @export
 */
const AgentVersionInEnvironmentApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.AgentVersionInEnvironmentApiFp)(configuration);
    return {
        /**
         *
         * @param {string} [agentId]
         * @param {string} [environmentId]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksGet(agentId, environmentId, skip, take, options) {
            return localVarFp.apiV1AgentsLinksGet(agentId, environmentId, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksIdDelete(id, options) {
            return localVarFp.apiV1AgentsLinksIdDelete(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksIdGet(id, options) {
            return localVarFp.apiV1AgentsLinksIdGet(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {AgentVersionInEnvironmentUpdateDTO} [agentVersionInEnvironmentUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksIdPatch(id, agentVersionInEnvironmentUpdateDTO, options) {
            return localVarFp.apiV1AgentsLinksIdPatch(id, agentVersionInEnvironmentUpdateDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {AgentVersionInEnvironmentCreateDTO} [agentVersionInEnvironmentCreateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsLinksPost(agentVersionInEnvironmentCreateDTO, options) {
            return localVarFp.apiV1AgentsLinksPost(agentVersionInEnvironmentCreateDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.AgentVersionInEnvironmentApiFactory = AgentVersionInEnvironmentApiFactory;
/**
 * AgentVersionInEnvironmentApi - object-oriented interface
 * @export
 * @class AgentVersionInEnvironmentApi
 * @extends {BaseAPI}
 */
class AgentVersionInEnvironmentApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} [agentId]
     * @param {string} [environmentId]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionInEnvironmentApi
     */
    apiV1AgentsLinksGet(agentId, environmentId, skip, take, options) {
        return (0, exports.AgentVersionInEnvironmentApiFp)(this.configuration).apiV1AgentsLinksGet(agentId, environmentId, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionInEnvironmentApi
     */
    apiV1AgentsLinksIdDelete(id, options) {
        return (0, exports.AgentVersionInEnvironmentApiFp)(this.configuration).apiV1AgentsLinksIdDelete(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionInEnvironmentApi
     */
    apiV1AgentsLinksIdGet(id, options) {
        return (0, exports.AgentVersionInEnvironmentApiFp)(this.configuration).apiV1AgentsLinksIdGet(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {AgentVersionInEnvironmentUpdateDTO} [agentVersionInEnvironmentUpdateDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionInEnvironmentApi
     */
    apiV1AgentsLinksIdPatch(id, agentVersionInEnvironmentUpdateDTO, options) {
        return (0, exports.AgentVersionInEnvironmentApiFp)(this.configuration).apiV1AgentsLinksIdPatch(id, agentVersionInEnvironmentUpdateDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {AgentVersionInEnvironmentCreateDTO} [agentVersionInEnvironmentCreateDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionInEnvironmentApi
     */
    apiV1AgentsLinksPost(agentVersionInEnvironmentCreateDTO, options) {
        return (0, exports.AgentVersionInEnvironmentApiFp)(this.configuration).apiV1AgentsLinksPost(agentVersionInEnvironmentCreateDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.AgentVersionInEnvironmentApi = AgentVersionInEnvironmentApi;
