// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/agent-version-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentVersionApi = exports.AgentVersionApiFactory = exports.AgentVersionApiFp = exports.AgentVersionApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * AgentVersionApi - axios parameter creator
 * @export
 */
const AgentVersionApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} [agentId]
         * @param {HiddenStateEnum} [hiddenState]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsGet: async (agentId, hiddenState, skip, take, options = {}) => {
            const localVarPath = `/api/v1/agents/versions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (agentId !== undefined) {
                localVarQueryParameter['agentId'] = agentId;
            }
            if (hiddenState !== undefined) {
                localVarQueryParameter['hiddenState'] = hiddenState;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsIdBinaryGet: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsVersionsIdBinaryGet', 'id', id);
            const localVarPath = `/api/v1/agents/versions/{id}/binary`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsIdGet: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsVersionsIdGet', 'id', id);
            const localVarPath = `/api/v1/agents/versions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {AgentVersionUpdateDTO} [agentVersionUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsIdPatch: async (id, agentVersionUpdateDTO, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsVersionsIdPatch', 'id', id);
            const localVarPath = `/api/v1/agents/versions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(agentVersionUpdateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} agentId
         * @param {File} body
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsPost: async (agentId, body, options = {}) => {
            // verify required parameter 'agentId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsVersionsPost', 'agentId', agentId);
            // verify required parameter 'body' is not null or undefined
            (0, common_1.assertParamExists)('apiV1AgentsVersionsPost', 'body', body);
            const localVarPath = `/api/v1/agents/versions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (agentId !== undefined) {
                localVarQueryParameter['agentId'] = agentId;
            }
            localVarHeaderParameter['Content-Type'] = 'application/octet-stream';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(body, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.AgentVersionApiAxiosParamCreator = AgentVersionApiAxiosParamCreator;
/**
 * AgentVersionApi - functional programming interface
 * @export
 */
const AgentVersionApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.AgentVersionApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} [agentId]
         * @param {HiddenStateEnum} [hiddenState]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsVersionsGet(agentId, hiddenState, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsVersionsGet(agentId, hiddenState, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionApi.apiV1AgentsVersionsGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsVersionsIdBinaryGet(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsVersionsIdBinaryGet(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionApi.apiV1AgentsVersionsIdBinaryGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsVersionsIdGet(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsVersionsIdGet(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionApi.apiV1AgentsVersionsIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {AgentVersionUpdateDTO} [agentVersionUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsVersionsIdPatch(id, agentVersionUpdateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsVersionsIdPatch(id, agentVersionUpdateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionApi.apiV1AgentsVersionsIdPatch']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} agentId
         * @param {File} body
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1AgentsVersionsPost(agentId, body, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1AgentsVersionsPost(agentId, body, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AgentVersionApi.apiV1AgentsVersionsPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.AgentVersionApiFp = AgentVersionApiFp;
/**
 * AgentVersionApi - factory interface
 * @export
 */
const AgentVersionApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.AgentVersionApiFp)(configuration);
    return {
        /**
         *
         * @param {string} [agentId]
         * @param {HiddenStateEnum} [hiddenState]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsGet(agentId, hiddenState, skip, take, options) {
            return localVarFp.apiV1AgentsVersionsGet(agentId, hiddenState, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsIdBinaryGet(id, options) {
            return localVarFp.apiV1AgentsVersionsIdBinaryGet(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsIdGet(id, options) {
            return localVarFp.apiV1AgentsVersionsIdGet(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {AgentVersionUpdateDTO} [agentVersionUpdateDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsIdPatch(id, agentVersionUpdateDTO, options) {
            return localVarFp.apiV1AgentsVersionsIdPatch(id, agentVersionUpdateDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} agentId
         * @param {File} body
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1AgentsVersionsPost(agentId, body, options) {
            return localVarFp.apiV1AgentsVersionsPost(agentId, body, options).then((request) => request(axios, basePath));
        },
    };
};
exports.AgentVersionApiFactory = AgentVersionApiFactory;
/**
 * AgentVersionApi - object-oriented interface
 * @export
 * @class AgentVersionApi
 * @extends {BaseAPI}
 */
class AgentVersionApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} [agentId]
     * @param {HiddenStateEnum} [hiddenState]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionApi
     */
    apiV1AgentsVersionsGet(agentId, hiddenState, skip, take, options) {
        return (0, exports.AgentVersionApiFp)(this.configuration).apiV1AgentsVersionsGet(agentId, hiddenState, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionApi
     */
    apiV1AgentsVersionsIdBinaryGet(id, options) {
        return (0, exports.AgentVersionApiFp)(this.configuration).apiV1AgentsVersionsIdBinaryGet(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionApi
     */
    apiV1AgentsVersionsIdGet(id, options) {
        return (0, exports.AgentVersionApiFp)(this.configuration).apiV1AgentsVersionsIdGet(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {AgentVersionUpdateDTO} [agentVersionUpdateDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionApi
     */
    apiV1AgentsVersionsIdPatch(id, agentVersionUpdateDTO, options) {
        return (0, exports.AgentVersionApiFp)(this.configuration).apiV1AgentsVersionsIdPatch(id, agentVersionUpdateDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} agentId
     * @param {File} body
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AgentVersionApi
     */
    apiV1AgentsVersionsPost(agentId, body, options) {
        return (0, exports.AgentVersionApiFp)(this.configuration).apiV1AgentsVersionsPost(agentId, body, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.AgentVersionApi = AgentVersionApi;
