// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/diagnostics-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiagnosticsApi = exports.DiagnosticsApiFactory = exports.DiagnosticsApiFp = exports.DiagnosticsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * DiagnosticsApi - axios parameter creator
 * @export
 */
const DiagnosticsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerInstancesGet: async (options = {}) => {
            const localVarPath = `/api/v1/diagnostics/customer/instances`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerLimitGet: async (options = {}) => {
            const localVarPath = `/api/v1/diagnostics/customer/limit`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueueLengthGet: async (options = {}) => {
            const localVarPath = `/api/v1/diagnostics/customer/queue/length`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueuesLengthsGet: async (options = {}) => {
            const localVarPath = `/api/v1/diagnostics/customer/queues/lengths`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet: async (groupId, applicationName, options = {}) => {
            // verify required parameter 'groupId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet', 'groupId', groupId);
            // verify required parameter 'applicationName' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet', 'applicationName', applicationName);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/application/{applicationName}/limit`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)))
                .replace(`{${"applicationName"}}`, encodeURIComponent(String(applicationName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet: async (groupId, applicationName, options = {}) => {
            // verify required parameter 'groupId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet', 'groupId', groupId);
            // verify required parameter 'applicationName' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet', 'applicationName', applicationName);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/application/{applicationName}/queue/length`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)))
                .replace(`{${"applicationName"}}`, encodeURIComponent(String(applicationName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdLimitGet: async (groupId, options = {}) => {
            // verify required parameter 'groupId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsGroupGroupIdLimitGet', 'groupId', groupId);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/limit`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdQueueLengthGet: async (groupId, options = {}) => {
            // verify required parameter 'groupId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsGroupGroupIdQueueLengthGet', 'groupId', groupId);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/queue/length`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet: async (instanceId, options = {}) => {
            // verify required parameter 'instanceId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}/all/queue/length`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdGet: async (instanceId, options = {}) => {
            // verify required parameter 'instanceId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsInstanceInstanceIdGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitGet: async (instanceId, options = {}) => {
            // verify required parameter 'instanceId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsInstanceInstanceIdLimitGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}/limit`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitsGet: async (instanceId, options = {}) => {
            // verify required parameter 'instanceId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1DiagnosticsInstanceInstanceIdLimitsGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}/limits`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsLimitsGet: async (options = {}) => {
            const localVarPath = `/api/v1/diagnostics/limits`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.DiagnosticsApiAxiosParamCreator = DiagnosticsApiAxiosParamCreator;
/**
 * DiagnosticsApi - functional programming interface
 * @export
 */
const DiagnosticsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.DiagnosticsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsCustomerInstancesGet(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsCustomerInstancesGet(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsCustomerInstancesGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsCustomerLimitGet(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsCustomerLimitGet(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsCustomerLimitGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsCustomerQueueLengthGet(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsCustomerQueueLengthGet(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsCustomerQueueLengthGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsCustomerQueuesLengthsGet(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsCustomerQueuesLengthsGet(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsCustomerQueuesLengthsGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsGroupGroupIdLimitGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsGroupGroupIdQueueLengthGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsInstanceInstanceIdGet(instanceId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdGet(instanceId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsInstanceInstanceIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsInstanceInstanceIdLimitGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsInstanceInstanceIdLimitsGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1DiagnosticsLimitsGet(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1DiagnosticsLimitsGet(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['DiagnosticsApi.apiV1DiagnosticsLimitsGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.DiagnosticsApiFp = DiagnosticsApiFp;
/**
 * DiagnosticsApi - factory interface
 * @export
 */
const DiagnosticsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.DiagnosticsApiFp)(configuration);
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerInstancesGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerInstancesGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerLimitGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerLimitGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueueLengthGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerQueueLengthGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueuesLengthsGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerQueuesLengthsGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsLimitsGet(options) {
            return localVarFp.apiV1DiagnosticsLimitsGet(options).then((request) => request(axios, basePath));
        },
    };
};
exports.DiagnosticsApiFactory = DiagnosticsApiFactory;
/**
 * DiagnosticsApi - object-oriented interface
 * @export
 * @class DiagnosticsApi
 * @extends {BaseAPI}
 */
class DiagnosticsApi extends base_1.BaseAPI {
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerInstancesGet(options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsCustomerInstancesGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerLimitGet(options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsCustomerLimitGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerQueueLengthGet(options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsCustomerQueueLengthGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerQueuesLengthsGet(options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsCustomerQueuesLengthsGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {string} applicationName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {string} applicationName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdGet(instanceId, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsInstanceInstanceIdGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsLimitsGet(options) {
        return (0, exports.DiagnosticsApiFp)(this.configuration).apiV1DiagnosticsLimitsGet(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.DiagnosticsApi = DiagnosticsApi;
