// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/export-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportApi = exports.ExportApiFactory = exports.ExportApiFp = exports.ExportApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ExportApi - axios parameter creator
 * @export
 */
const ExportApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {number} [maxColumn]
         * @param {number} [maxObjectFlatterLevel]
         * @param {boolean} [enableArrayFlatter]
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsCsvFilePost: async (maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options = {}) => {
            const localVarPath = `/api/v1/export-jobs/csv/file`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (maxColumn !== undefined) {
                localVarQueryParameter['maxColumn'] = maxColumn;
            }
            if (maxObjectFlatterLevel !== undefined) {
                localVarQueryParameter['maxObjectFlatterLevel'] = maxObjectFlatterLevel;
            }
            if (enableArrayFlatter !== undefined) {
                localVarQueryParameter['enableArrayFlatter'] = enableArrayFlatter;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(getOnlineProfilerExport, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {GetOnlineProfilerFilters} [getOnlineProfilerFilters]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsFiltersPost: async (getOnlineProfilerFilters, options = {}) => {
            const localVarPath = `/api/v1/export-jobs/filters`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(getOnlineProfilerFilters, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsJsonFilePost: async (getOnlineProfilerExport, options = {}) => {
            const localVarPath = `/api/v1/export-jobs/json/file`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(getOnlineProfilerExport, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsJsonPost: async (skip, take, getOnlineProfilerExport, options = {}) => {
            const localVarPath = `/api/v1/export-jobs/json`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(getOnlineProfilerExport, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ExportApiAxiosParamCreator = ExportApiAxiosParamCreator;
/**
 * ExportApi - functional programming interface
 * @export
 */
const ExportApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ExportApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {number} [maxColumn]
         * @param {number} [maxObjectFlatterLevel]
         * @param {boolean} [enableArrayFlatter]
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ExportJobsCsvFilePost(maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ExportJobsCsvFilePost(maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ExportApi.apiV1ExportJobsCsvFilePost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {GetOnlineProfilerFilters} [getOnlineProfilerFilters]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ExportJobsFiltersPost(getOnlineProfilerFilters, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ExportJobsFiltersPost(getOnlineProfilerFilters, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ExportApi.apiV1ExportJobsFiltersPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ExportJobsJsonFilePost(getOnlineProfilerExport, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ExportJobsJsonFilePost(getOnlineProfilerExport, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ExportApi.apiV1ExportJobsJsonFilePost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ExportJobsJsonPost(skip, take, getOnlineProfilerExport, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ExportJobsJsonPost(skip, take, getOnlineProfilerExport, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ExportApi.apiV1ExportJobsJsonPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.ExportApiFp = ExportApiFp;
/**
 * ExportApi - factory interface
 * @export
 */
const ExportApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ExportApiFp)(configuration);
    return {
        /**
         *
         * @param {number} [maxColumn]
         * @param {number} [maxObjectFlatterLevel]
         * @param {boolean} [enableArrayFlatter]
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsCsvFilePost(maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options) {
            return localVarFp.apiV1ExportJobsCsvFilePost(maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {GetOnlineProfilerFilters} [getOnlineProfilerFilters]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsFiltersPost(getOnlineProfilerFilters, options) {
            return localVarFp.apiV1ExportJobsFiltersPost(getOnlineProfilerFilters, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsJsonFilePost(getOnlineProfilerExport, options) {
            return localVarFp.apiV1ExportJobsJsonFilePost(getOnlineProfilerExport, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ExportJobsJsonPost(skip, take, getOnlineProfilerExport, options) {
            return localVarFp.apiV1ExportJobsJsonPost(skip, take, getOnlineProfilerExport, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ExportApiFactory = ExportApiFactory;
/**
 * ExportApi - object-oriented interface
 * @export
 * @class ExportApi
 * @extends {BaseAPI}
 */
class ExportApi extends base_1.BaseAPI {
    /**
     *
     * @param {number} [maxColumn]
     * @param {number} [maxObjectFlatterLevel]
     * @param {boolean} [enableArrayFlatter]
     * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ExportApi
     */
    apiV1ExportJobsCsvFilePost(maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options) {
        return (0, exports.ExportApiFp)(this.configuration).apiV1ExportJobsCsvFilePost(maxColumn, maxObjectFlatterLevel, enableArrayFlatter, getOnlineProfilerExport, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {GetOnlineProfilerFilters} [getOnlineProfilerFilters]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ExportApi
     */
    apiV1ExportJobsFiltersPost(getOnlineProfilerFilters, options) {
        return (0, exports.ExportApiFp)(this.configuration).apiV1ExportJobsFiltersPost(getOnlineProfilerFilters, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ExportApi
     */
    apiV1ExportJobsJsonFilePost(getOnlineProfilerExport, options) {
        return (0, exports.ExportApiFp)(this.configuration).apiV1ExportJobsJsonFilePost(getOnlineProfilerExport, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {number} [skip]
     * @param {number} [take]
     * @param {GetOnlineProfilerExport} [getOnlineProfilerExport]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ExportApi
     */
    apiV1ExportJobsJsonPost(skip, take, getOnlineProfilerExport, options) {
        return (0, exports.ExportApiFp)(this.configuration).apiV1ExportJobsJsonPost(skip, take, getOnlineProfilerExport, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ExportApi = ExportApi;
