// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/intent-generator-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntentGeneratorApi = exports.IntentGeneratorApiFactory = exports.IntentGeneratorApiFp = exports.IntentGeneratorApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * IntentGeneratorApi - axios parameter creator
 * @export
 */
const IntentGeneratorApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} name
         * @param {IntentGeneratorDTO} [intentGeneratorDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1GeneratorIntentNamePost: async (name, intentGeneratorDTO, options = {}) => {
            // verify required parameter 'name' is not null or undefined
            (0, common_1.assertParamExists)('apiV1GeneratorIntentNamePost', 'name', name);
            const localVarPath = `/api/v1/generator/intent/{name}`
                .replace(`{${"name"}}`, encodeURIComponent(String(name)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(intentGeneratorDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.IntentGeneratorApiAxiosParamCreator = IntentGeneratorApiAxiosParamCreator;
/**
 * IntentGeneratorApi - functional programming interface
 * @export
 */
const IntentGeneratorApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.IntentGeneratorApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {IntentGeneratorDTO} [intentGeneratorDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1GeneratorIntentNamePost(name, intentGeneratorDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1GeneratorIntentNamePost(name, intentGeneratorDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['IntentGeneratorApi.apiV1GeneratorIntentNamePost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.IntentGeneratorApiFp = IntentGeneratorApiFp;
/**
 * IntentGeneratorApi - factory interface
 * @export
 */
const IntentGeneratorApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.IntentGeneratorApiFp)(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {IntentGeneratorDTO} [intentGeneratorDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1GeneratorIntentNamePost(name, intentGeneratorDTO, options) {
            return localVarFp.apiV1GeneratorIntentNamePost(name, intentGeneratorDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.IntentGeneratorApiFactory = IntentGeneratorApiFactory;
/**
 * IntentGeneratorApi - object-oriented interface
 * @export
 * @class IntentGeneratorApi
 * @extends {BaseAPI}
 */
class IntentGeneratorApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} name
     * @param {IntentGeneratorDTO} [intentGeneratorDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof IntentGeneratorApi
     */
    apiV1GeneratorIntentNamePost(name, intentGeneratorDTO, options) {
        return (0, exports.IntentGeneratorApiFp)(this.configuration).apiV1GeneratorIntentNamePost(name, intentGeneratorDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.IntentGeneratorApi = IntentGeneratorApi;
