// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/package-permission-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackagePermissionApi = exports.PackagePermissionApiFactory = exports.PackagePermissionApiFp = exports.PackagePermissionApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PackagePermissionApi - axios parameter creator
 * @export
 */
const PackagePermissionApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} packageDescriptionId
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionByNamePackageDescriptionIdListGet: async (packageDescriptionId, skip, take, options = {}) => {
            // verify required parameter 'packageDescriptionId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesPermissionByNamePackageDescriptionIdListGet', 'packageDescriptionId', packageDescriptionId);
            const localVarPath = `/api/v1/packages/permission/by-name/{packageDescriptionId}/list`
                .replace(`{${"packageDescriptionId"}}`, encodeURIComponent(String(packageDescriptionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {PackagePermissionsDeleteDTO} packagePermissionsDeleteDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionDelete: async (packagePermissionsDeleteDTO, options = {}) => {
            // verify required parameter 'packagePermissionsDeleteDTO' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesPermissionDelete', 'packagePermissionsDeleteDTO', packagePermissionsDeleteDTO);
            const localVarPath = `/api/v1/packages/permission`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(packagePermissionsDeleteDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} packageName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionGet: async (packageName, options = {}) => {
            // verify required parameter 'packageName' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesPermissionGet', 'packageName', packageName);
            const localVarPath = `/api/v1/packages/permission`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (packageName !== undefined) {
                localVarQueryParameter['packageName'] = packageName;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} packageName
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionListGet: async (packageName, skip, take, options = {}) => {
            // verify required parameter 'packageName' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesPermissionListGet', 'packageName', packageName);
            const localVarPath = `/api/v1/packages/permission/list`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (packageName !== undefined) {
                localVarQueryParameter['packageName'] = packageName;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionOwnListGet: async (skip, take, options = {}) => {
            const localVarPath = `/api/v1/packages/permission/own/list`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} packageDescriptionId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionPackageDescriptionIdGet: async (packageDescriptionId, options = {}) => {
            // verify required parameter 'packageDescriptionId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesPermissionPackageDescriptionIdGet', 'packageDescriptionId', packageDescriptionId);
            const localVarPath = `/api/v1/packages/permission/{packageDescriptionId}`
                .replace(`{${"packageDescriptionId"}}`, encodeURIComponent(String(packageDescriptionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {PackagePermissionsCreateOrUpdateDTO} packagePermissionsCreateOrUpdateDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionPost: async (packagePermissionsCreateOrUpdateDTO, options = {}) => {
            // verify required parameter 'packagePermissionsCreateOrUpdateDTO' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesPermissionPost', 'packagePermissionsCreateOrUpdateDTO', packagePermissionsCreateOrUpdateDTO);
            const localVarPath = `/api/v1/packages/permission`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(packagePermissionsCreateOrUpdateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.PackagePermissionApiAxiosParamCreator = PackagePermissionApiAxiosParamCreator;
/**
 * PackagePermissionApi - functional programming interface
 * @export
 */
const PackagePermissionApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PackagePermissionApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} packageDescriptionId
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageDescriptionId, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageDescriptionId, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionByNamePackageDescriptionIdListGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {PackagePermissionsDeleteDTO} packagePermissionsDeleteDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionDelete(packagePermissionsDeleteDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionDelete(packagePermissionsDeleteDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionDelete']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} packageName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionGet(packageName, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionGet(packageName, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} packageName
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionListGet(packageName, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionListGet(packageName, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionListGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionOwnListGet(skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionOwnListGet(skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionOwnListGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} packageDescriptionId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionPackageDescriptionIdGet(packageDescriptionId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionPackageDescriptionIdGet(packageDescriptionId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionPackageDescriptionIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {PackagePermissionsCreateOrUpdateDTO} packagePermissionsCreateOrUpdateDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesPermissionPost(packagePermissionsCreateOrUpdateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesPermissionPost(packagePermissionsCreateOrUpdateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackagePermissionApi.apiV1PackagesPermissionPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.PackagePermissionApiFp = PackagePermissionApiFp;
/**
 * PackagePermissionApi - factory interface
 * @export
 */
const PackagePermissionApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PackagePermissionApiFp)(configuration);
    return {
        /**
         *
         * @param {string} packageDescriptionId
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageDescriptionId, skip, take, options) {
            return localVarFp.apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageDescriptionId, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {PackagePermissionsDeleteDTO} packagePermissionsDeleteDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionDelete(packagePermissionsDeleteDTO, options) {
            return localVarFp.apiV1PackagesPermissionDelete(packagePermissionsDeleteDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} packageName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionGet(packageName, options) {
            return localVarFp.apiV1PackagesPermissionGet(packageName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} packageName
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionListGet(packageName, skip, take, options) {
            return localVarFp.apiV1PackagesPermissionListGet(packageName, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionOwnListGet(skip, take, options) {
            return localVarFp.apiV1PackagesPermissionOwnListGet(skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} packageDescriptionId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionPackageDescriptionIdGet(packageDescriptionId, options) {
            return localVarFp.apiV1PackagesPermissionPackageDescriptionIdGet(packageDescriptionId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {PackagePermissionsCreateOrUpdateDTO} packagePermissionsCreateOrUpdateDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesPermissionPost(packagePermissionsCreateOrUpdateDTO, options) {
            return localVarFp.apiV1PackagesPermissionPost(packagePermissionsCreateOrUpdateDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.PackagePermissionApiFactory = PackagePermissionApiFactory;
/**
 * PackagePermissionApi - object-oriented interface
 * @export
 * @class PackagePermissionApi
 * @extends {BaseAPI}
 */
class PackagePermissionApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} packageDescriptionId
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageDescriptionId, skip, take, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionByNamePackageDescriptionIdListGet(packageDescriptionId, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {PackagePermissionsDeleteDTO} packagePermissionsDeleteDTO
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionDelete(packagePermissionsDeleteDTO, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionDelete(packagePermissionsDeleteDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} packageName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionGet(packageName, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionGet(packageName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} packageName
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionListGet(packageName, skip, take, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionListGet(packageName, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionOwnListGet(skip, take, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionOwnListGet(skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} packageDescriptionId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionPackageDescriptionIdGet(packageDescriptionId, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionPackageDescriptionIdGet(packageDescriptionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {PackagePermissionsCreateOrUpdateDTO} packagePermissionsCreateOrUpdateDTO
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackagePermissionApi
     */
    apiV1PackagesPermissionPost(packagePermissionsCreateOrUpdateDTO, options) {
        return (0, exports.PackagePermissionApiFp)(this.configuration).apiV1PackagesPermissionPost(packagePermissionsCreateOrUpdateDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.PackagePermissionApi = PackagePermissionApi;
