// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/package-dependency-resolver-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageDependencyResolverApi = exports.PackageDependencyResolverApiFactory = exports.PackageDependencyResolverApiFp = exports.PackageDependencyResolverApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PackageDependencyResolverApi - axios parameter creator
 * @export
 */
const PackageDependencyResolverApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {PackageDependencyResolveDTO} packageDependencyResolveDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesResolverPost: async (packageDependencyResolveDTO, options = {}) => {
            // verify required parameter 'packageDependencyResolveDTO' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesResolverPost', 'packageDependencyResolveDTO', packageDependencyResolveDTO);
            const localVarPath = `/api/v1/packages/resolver`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(packageDependencyResolveDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.PackageDependencyResolverApiAxiosParamCreator = PackageDependencyResolverApiAxiosParamCreator;
/**
 * PackageDependencyResolverApi - functional programming interface
 * @export
 */
const PackageDependencyResolverApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PackageDependencyResolverApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {PackageDependencyResolveDTO} packageDependencyResolveDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesResolverPost(packageDependencyResolveDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesResolverPost(packageDependencyResolveDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackageDependencyResolverApi.apiV1PackagesResolverPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.PackageDependencyResolverApiFp = PackageDependencyResolverApiFp;
/**
 * PackageDependencyResolverApi - factory interface
 * @export
 */
const PackageDependencyResolverApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PackageDependencyResolverApiFp)(configuration);
    return {
        /**
         *
         * @param {PackageDependencyResolveDTO} packageDependencyResolveDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesResolverPost(packageDependencyResolveDTO, options) {
            return localVarFp.apiV1PackagesResolverPost(packageDependencyResolveDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.PackageDependencyResolverApiFactory = PackageDependencyResolverApiFactory;
/**
 * PackageDependencyResolverApi - object-oriented interface
 * @export
 * @class PackageDependencyResolverApi
 * @extends {BaseAPI}
 */
class PackageDependencyResolverApi extends base_1.BaseAPI {
    /**
     *
     * @param {PackageDependencyResolveDTO} packageDependencyResolveDTO
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackageDependencyResolverApi
     */
    apiV1PackagesResolverPost(packageDependencyResolveDTO, options) {
        return (0, exports.PackageDependencyResolverApiFp)(this.configuration).apiV1PackagesResolverPost(packageDependencyResolveDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.PackageDependencyResolverApi = PackageDependencyResolverApi;
