// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/package-description-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageDescriptionApi = exports.PackageDescriptionApiFactory = exports.PackageDescriptionApiFp = exports.PackageDescriptionApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PackageDescriptionApi - axios parameter creator
 * @export
 */
const PackageDescriptionApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} packageName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionGet: async (packageName, options = {}) => {
            // verify required parameter 'packageName' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesDescriptionGet', 'packageName', packageName);
            const localVarPath = `/api/v1/packages/description`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (packageName !== undefined) {
                localVarQueryParameter['packageName'] = packageName;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {PackageContentType} [contentType]
         * @param {string} [ownerId]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionListGet: async (contentType, ownerId, skip, take, options = {}) => {
            const localVarPath = `/api/v1/packages/description/list`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (contentType !== undefined) {
                localVarQueryParameter['contentType'] = contentType;
            }
            if (ownerId !== undefined) {
                localVarQueryParameter['ownerId'] = ownerId;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} packageDescriptionId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionPackageDescriptionIdGet: async (packageDescriptionId, options = {}) => {
            // verify required parameter 'packageDescriptionId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesDescriptionPackageDescriptionIdGet', 'packageDescriptionId', packageDescriptionId);
            const localVarPath = `/api/v1/packages/description/{packageDescriptionId}`
                .replace(`{${"packageDescriptionId"}}`, encodeURIComponent(String(packageDescriptionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {PackageDescriptionCreateOrUpdateDTO} packageDescriptionCreateOrUpdateDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionPost: async (packageDescriptionCreateOrUpdateDTO, options = {}) => {
            // verify required parameter 'packageDescriptionCreateOrUpdateDTO' is not null or undefined
            (0, common_1.assertParamExists)('apiV1PackagesDescriptionPost', 'packageDescriptionCreateOrUpdateDTO', packageDescriptionCreateOrUpdateDTO);
            const localVarPath = `/api/v1/packages/description`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(packageDescriptionCreateOrUpdateDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.PackageDescriptionApiAxiosParamCreator = PackageDescriptionApiAxiosParamCreator;
/**
 * PackageDescriptionApi - functional programming interface
 * @export
 */
const PackageDescriptionApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PackageDescriptionApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} packageName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesDescriptionGet(packageName, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesDescriptionGet(packageName, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackageDescriptionApi.apiV1PackagesDescriptionGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {PackageContentType} [contentType]
         * @param {string} [ownerId]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesDescriptionListGet(contentType, ownerId, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesDescriptionListGet(contentType, ownerId, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackageDescriptionApi.apiV1PackagesDescriptionListGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} packageDescriptionId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesDescriptionPackageDescriptionIdGet(packageDescriptionId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesDescriptionPackageDescriptionIdGet(packageDescriptionId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackageDescriptionApi.apiV1PackagesDescriptionPackageDescriptionIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {PackageDescriptionCreateOrUpdateDTO} packageDescriptionCreateOrUpdateDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1PackagesDescriptionPost(packageDescriptionCreateOrUpdateDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1PackagesDescriptionPost(packageDescriptionCreateOrUpdateDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['PackageDescriptionApi.apiV1PackagesDescriptionPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.PackageDescriptionApiFp = PackageDescriptionApiFp;
/**
 * PackageDescriptionApi - factory interface
 * @export
 */
const PackageDescriptionApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PackageDescriptionApiFp)(configuration);
    return {
        /**
         *
         * @param {string} packageName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionGet(packageName, options) {
            return localVarFp.apiV1PackagesDescriptionGet(packageName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {PackageContentType} [contentType]
         * @param {string} [ownerId]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionListGet(contentType, ownerId, skip, take, options) {
            return localVarFp.apiV1PackagesDescriptionListGet(contentType, ownerId, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} packageDescriptionId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionPackageDescriptionIdGet(packageDescriptionId, options) {
            return localVarFp.apiV1PackagesDescriptionPackageDescriptionIdGet(packageDescriptionId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {PackageDescriptionCreateOrUpdateDTO} packageDescriptionCreateOrUpdateDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1PackagesDescriptionPost(packageDescriptionCreateOrUpdateDTO, options) {
            return localVarFp.apiV1PackagesDescriptionPost(packageDescriptionCreateOrUpdateDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.PackageDescriptionApiFactory = PackageDescriptionApiFactory;
/**
 * PackageDescriptionApi - object-oriented interface
 * @export
 * @class PackageDescriptionApi
 * @extends {BaseAPI}
 */
class PackageDescriptionApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} packageName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackageDescriptionApi
     */
    apiV1PackagesDescriptionGet(packageName, options) {
        return (0, exports.PackageDescriptionApiFp)(this.configuration).apiV1PackagesDescriptionGet(packageName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {PackageContentType} [contentType]
     * @param {string} [ownerId]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackageDescriptionApi
     */
    apiV1PackagesDescriptionListGet(contentType, ownerId, skip, take, options) {
        return (0, exports.PackageDescriptionApiFp)(this.configuration).apiV1PackagesDescriptionListGet(contentType, ownerId, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} packageDescriptionId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackageDescriptionApi
     */
    apiV1PackagesDescriptionPackageDescriptionIdGet(packageDescriptionId, options) {
        return (0, exports.PackageDescriptionApiFp)(this.configuration).apiV1PackagesDescriptionPackageDescriptionIdGet(packageDescriptionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {PackageDescriptionCreateOrUpdateDTO} packageDescriptionCreateOrUpdateDTO
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PackageDescriptionApi
     */
    apiV1PackagesDescriptionPost(packageDescriptionCreateOrUpdateDTO, options) {
        return (0, exports.PackageDescriptionApiFp)(this.configuration).apiV1PackagesDescriptionPost(packageDescriptionCreateOrUpdateDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.PackageDescriptionApi = PackageDescriptionApi;
