// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/record-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordApi = exports.RecordApiFactory = exports.RecordApiFp = exports.RecordApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * RecordApi - axios parameter creator
 * @export
 */
const RecordApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} recordId
         * @param {boolean} [inBrowser]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RecordsRecordIdGet: async (recordId, inBrowser, options = {}) => {
            // verify required parameter 'recordId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1RecordsRecordIdGet', 'recordId', recordId);
            const localVarPath = `/api/v1/records/{recordId}`
                .replace(`{${"recordId"}}`, encodeURIComponent(String(recordId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (inBrowser !== undefined) {
                localVarQueryParameter['inBrowser'] = inBrowser;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.RecordApiAxiosParamCreator = RecordApiAxiosParamCreator;
/**
 * RecordApi - functional programming interface
 * @export
 */
const RecordApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.RecordApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} recordId
         * @param {boolean} [inBrowser]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1RecordsRecordIdGet(recordId, inBrowser, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1RecordsRecordIdGet(recordId, inBrowser, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['RecordApi.apiV1RecordsRecordIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.RecordApiFp = RecordApiFp;
/**
 * RecordApi - factory interface
 * @export
 */
const RecordApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.RecordApiFp)(configuration);
    return {
        /**
         *
         * @param {string} recordId
         * @param {boolean} [inBrowser]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RecordsRecordIdGet(recordId, inBrowser, options) {
            return localVarFp.apiV1RecordsRecordIdGet(recordId, inBrowser, options).then((request) => request(axios, basePath));
        },
    };
};
exports.RecordApiFactory = RecordApiFactory;
/**
 * RecordApi - object-oriented interface
 * @export
 * @class RecordApi
 * @extends {BaseAPI}
 */
class RecordApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} recordId
     * @param {boolean} [inBrowser]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof RecordApi
     */
    apiV1RecordsRecordIdGet(recordId, inBrowser, options) {
        return (0, exports.RecordApiFp)(this.configuration).apiV1RecordsRecordIdGet(recordId, inBrowser, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.RecordApi = RecordApi;
