// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/register-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisterApi = exports.RegisterApiFactory = exports.RegisterApiFp = exports.RegisterApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * RegisterApi - axios parameter creator
 * @export
 */
const RegisterApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} name
         * @param {number} [maxConcurrency]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RegisterPost: async (name, maxConcurrency, options = {}) => {
            // verify required parameter 'name' is not null or undefined
            (0, common_1.assertParamExists)('apiV1RegisterPost', 'name', name);
            const localVarPath = `/api/v1/register`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (maxConcurrency !== undefined) {
                localVarQueryParameter['maxConcurrency'] = maxConcurrency;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.RegisterApiAxiosParamCreator = RegisterApiAxiosParamCreator;
/**
 * RegisterApi - functional programming interface
 * @export
 */
const RegisterApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.RegisterApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {number} [maxConcurrency]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1RegisterPost(name, maxConcurrency, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1RegisterPost(name, maxConcurrency, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['RegisterApi.apiV1RegisterPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.RegisterApiFp = RegisterApiFp;
/**
 * RegisterApi - factory interface
 * @export
 */
const RegisterApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.RegisterApiFp)(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {number} [maxConcurrency]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RegisterPost(name, maxConcurrency, options) {
            return localVarFp.apiV1RegisterPost(name, maxConcurrency, options).then((request) => request(axios, basePath));
        },
    };
};
exports.RegisterApiFactory = RegisterApiFactory;
/**
 * RegisterApi - object-oriented interface
 * @export
 * @class RegisterApi
 * @extends {BaseAPI}
 */
class RegisterApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} name
     * @param {number} [maxConcurrency]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof RegisterApi
     */
    apiV1RegisterPost(name, maxConcurrency, options) {
        return (0, exports.RegisterApiFp)(this.configuration).apiV1RegisterPost(name, maxConcurrency, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.RegisterApi = RegisterApi;
