// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/session-log-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionLogApi = exports.SessionLogApiFactory = exports.SessionLogApiFp = exports.SessionLogApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SessionLogApi - axios parameter creator
 * @export
 */
const SessionLogApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsDebugJobIdGet: async (jobId, options = {}) => {
            // verify required parameter 'jobId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1LogsDebugJobIdGet', 'jobId', jobId);
            const localVarPath = `/api/v1/logs/debug/{jobId}`
                .replace(`{${"jobId"}}`, encodeURIComponent(String(jobId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} [fromCompletedTime]
         * @param {string} [toCompletedTime]
         * @param {TaskStatus} [taskStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsEntitiesGet: async (fromCompletedTime, toCompletedTime, taskStatus, skip, take, options = {}) => {
            const localVarPath = `/api/v1/logs/entities`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (fromCompletedTime !== undefined) {
                localVarQueryParameter['fromCompletedTime'] = (fromCompletedTime instanceof Date) ?
                    fromCompletedTime.toISOString() :
                    fromCompletedTime;
            }
            if (toCompletedTime !== undefined) {
                localVarQueryParameter['toCompletedTime'] = (toCompletedTime instanceof Date) ?
                    toCompletedTime.toISOString() :
                    toCompletedTime;
            }
            if (taskStatus !== undefined) {
                localVarQueryParameter['taskStatus'] = taskStatus;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsEntityJobIdGet: async (jobId, options = {}) => {
            // verify required parameter 'jobId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1LogsEntityJobIdGet', 'jobId', jobId);
            const localVarPath = `/api/v1/logs/entity/{jobId}`
                .replace(`{${"jobId"}}`, encodeURIComponent(String(jobId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.SessionLogApiAxiosParamCreator = SessionLogApiAxiosParamCreator;
/**
 * SessionLogApi - functional programming interface
 * @export
 */
const SessionLogApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.SessionLogApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1LogsDebugJobIdGet(jobId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1LogsDebugJobIdGet(jobId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SessionLogApi.apiV1LogsDebugJobIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} [fromCompletedTime]
         * @param {string} [toCompletedTime]
         * @param {TaskStatus} [taskStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1LogsEntitiesGet(fromCompletedTime, toCompletedTime, taskStatus, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1LogsEntitiesGet(fromCompletedTime, toCompletedTime, taskStatus, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SessionLogApi.apiV1LogsEntitiesGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1LogsEntityJobIdGet(jobId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1LogsEntityJobIdGet(jobId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SessionLogApi.apiV1LogsEntityJobIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.SessionLogApiFp = SessionLogApiFp;
/**
 * SessionLogApi - factory interface
 * @export
 */
const SessionLogApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.SessionLogApiFp)(configuration);
    return {
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsDebugJobIdGet(jobId, options) {
            return localVarFp.apiV1LogsDebugJobIdGet(jobId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} [fromCompletedTime]
         * @param {string} [toCompletedTime]
         * @param {TaskStatus} [taskStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsEntitiesGet(fromCompletedTime, toCompletedTime, taskStatus, skip, take, options) {
            return localVarFp.apiV1LogsEntitiesGet(fromCompletedTime, toCompletedTime, taskStatus, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsEntityJobIdGet(jobId, options) {
            return localVarFp.apiV1LogsEntityJobIdGet(jobId, options).then((request) => request(axios, basePath));
        },
    };
};
exports.SessionLogApiFactory = SessionLogApiFactory;
/**
 * SessionLogApi - object-oriented interface
 * @export
 * @class SessionLogApi
 * @extends {BaseAPI}
 */
class SessionLogApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} jobId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SessionLogApi
     */
    apiV1LogsDebugJobIdGet(jobId, options) {
        return (0, exports.SessionLogApiFp)(this.configuration).apiV1LogsDebugJobIdGet(jobId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} [fromCompletedTime]
     * @param {string} [toCompletedTime]
     * @param {TaskStatus} [taskStatus]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SessionLogApi
     */
    apiV1LogsEntitiesGet(fromCompletedTime, toCompletedTime, taskStatus, skip, take, options) {
        return (0, exports.SessionLogApiFp)(this.configuration).apiV1LogsEntitiesGet(fromCompletedTime, toCompletedTime, taskStatus, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} jobId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SessionLogApi
     */
    apiV1LogsEntityJobIdGet(jobId, options) {
        return (0, exports.SessionLogApiFp)(this.configuration).apiV1LogsEntityJobIdGet(jobId, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SessionLogApi = SessionLogApi;
