// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/core/api/sipendpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SIPEndpointApi = exports.SIPEndpointApiFactory = exports.SIPEndpointApiFp = exports.SIPEndpointApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SIPEndpointApi - axios parameter creator
 * @export
 */
const SIPEndpointApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointByNameGet: async (name, options = {}) => {
            // verify required parameter 'name' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointByNameGet', 'name', name);
            const localVarPath = `/api/v1/sip/endpoint/by-name`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} alias
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointCapacityByAliasGet: async (alias, options = {}) => {
            // verify required parameter 'alias' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointCapacityByAliasGet', 'alias', alias);
            const localVarPath = `/api/v1/sip/endpoint/capacity/by-alias`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (alias !== undefined) {
                localVarQueryParameter['alias'] = alias;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} endpointId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointCapacityEndpointIdGet: async (endpointId, options = {}) => {
            // verify required parameter 'endpointId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointCapacityEndpointIdGet', 'endpointId', endpointId);
            const localVarPath = `/api/v1/sip/endpoint/capacity/{endpointId}`
                .replace(`{${"endpointId"}}`, encodeURIComponent(String(endpointId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointGet: async (skip, take, options = {}) => {
            const localVarPath = `/api/v1/sip/endpoint`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdDelete: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointIdDelete', 'id', id);
            const localVarPath = `/api/v1/sip/endpoint/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdGet: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointIdGet', 'id', id);
            const localVarPath = `/api/v1/sip/endpoint/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdPatch: async (id, sIPEndpointPatchDTO, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointIdPatch', 'id', id);
            const localVarPath = `/api/v1/sip/endpoint/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(sIPEndpointPatchDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointPost: async (sIPEndpointAddDTO, options = {}) => {
            const localVarPath = `/api/v1/sip/endpoint`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(sIPEndpointAddDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} endpointId
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointReportsEndpointIdGet: async (endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options = {}) => {
            // verify required parameter 'endpointId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointReportsEndpointIdGet', 'endpointId', endpointId);
            const localVarPath = `/api/v1/sip/endpoint/reports/{endpointId}`
                .replace(`{${"endpointId"}}`, encodeURIComponent(String(endpointId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (fromCreatedTime !== undefined) {
                localVarQueryParameter['fromCreatedTime'] = (fromCreatedTime instanceof Date) ?
                    fromCreatedTime.toISOString() :
                    fromCreatedTime;
            }
            if (toCreatedTime !== undefined) {
                localVarQueryParameter['toCreatedTime'] = (toCreatedTime instanceof Date) ?
                    toCreatedTime.toISOString() :
                    toCreatedTime;
            }
            if (fromStatus !== undefined) {
                localVarQueryParameter['fromStatus'] = fromStatus;
            }
            if (toStatus !== undefined) {
                localVarQueryParameter['toStatus'] = toStatus;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointReportsGet: async (fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options = {}) => {
            const localVarPath = `/api/v1/sip/endpoint/reports`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (fromCreatedTime !== undefined) {
                localVarQueryParameter['fromCreatedTime'] = (fromCreatedTime instanceof Date) ?
                    fromCreatedTime.toISOString() :
                    fromCreatedTime;
            }
            if (toCreatedTime !== undefined) {
                localVarQueryParameter['toCreatedTime'] = (toCreatedTime instanceof Date) ?
                    toCreatedTime.toISOString() :
                    toCreatedTime;
            }
            if (fromStatus !== undefined) {
                localVarQueryParameter['fromStatus'] = fromStatus;
            }
            if (toStatus !== undefined) {
                localVarQueryParameter['toStatus'] = toStatus;
            }
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointReportsStatusGet: async (fromCreatedTime, toCreatedTime, fromStatus, toStatus, options = {}) => {
            const localVarPath = `/api/v1/sip/endpoint/reports/status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (fromCreatedTime !== undefined) {
                localVarQueryParameter['fromCreatedTime'] = (fromCreatedTime instanceof Date) ?
                    fromCreatedTime.toISOString() :
                    fromCreatedTime;
            }
            if (toCreatedTime !== undefined) {
                localVarQueryParameter['toCreatedTime'] = (toCreatedTime instanceof Date) ?
                    toCreatedTime.toISOString() :
                    toCreatedTime;
            }
            if (fromStatus !== undefined) {
                localVarQueryParameter['fromStatus'] = fromStatus;
            }
            if (toStatus !== undefined) {
                localVarQueryParameter['toStatus'] = toStatus;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} alias
         * @param {number} [errorsThreshold]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointStatusByAliasGet: async (alias, errorsThreshold, options = {}) => {
            // verify required parameter 'alias' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointStatusByAliasGet', 'alias', alias);
            const localVarPath = `/api/v1/sip/endpoint/status/by-alias`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (alias !== undefined) {
                localVarQueryParameter['alias'] = alias;
            }
            if (errorsThreshold !== undefined) {
                localVarQueryParameter['errorsThreshold'] = errorsThreshold;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} endpointId
         * @param {number} [errorsThreshold]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointStatusEndpointIdGet: async (endpointId, errorsThreshold, options = {}) => {
            // verify required parameter 'endpointId' is not null or undefined
            (0, common_1.assertParamExists)('apiV1SipEndpointStatusEndpointIdGet', 'endpointId', endpointId);
            const localVarPath = `/api/v1/sip/endpoint/status/{endpointId}`
                .replace(`{${"endpointId"}}`, encodeURIComponent(String(endpointId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (errorsThreshold !== undefined) {
                localVarQueryParameter['errorsThreshold'] = errorsThreshold;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.SIPEndpointApiAxiosParamCreator = SIPEndpointApiAxiosParamCreator;
/**
 * SIPEndpointApi - functional programming interface
 * @export
 */
const SIPEndpointApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.SIPEndpointApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointByNameGet(name, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointByNameGet(name, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointByNameGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} alias
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointCapacityByAliasGet(alias, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointCapacityByAliasGet(alias, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointCapacityByAliasGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} endpointId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointCapacityEndpointIdGet(endpointId, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointCapacityEndpointIdGet(endpointId, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointCapacityEndpointIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointGet(skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointGet(skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointIdDelete(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointIdDelete(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointIdDelete']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointIdGet(id, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointIdGet(id, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} id
         * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointIdPatch']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointPost(sIPEndpointAddDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointPost(sIPEndpointAddDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} endpointId
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointReportsEndpointIdGet(endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointReportsEndpointIdGet(endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointReportsEndpointIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointReportsGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointReportsGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointReportsGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointReportsStatusGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointReportsStatusGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointReportsStatusGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} alias
         * @param {number} [errorsThreshold]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointStatusByAliasGet(alias, errorsThreshold, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointStatusByAliasGet(alias, errorsThreshold, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointStatusByAliasGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} endpointId
         * @param {number} [errorsThreshold]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointStatusEndpointIdGet(endpointId, errorsThreshold, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointStatusEndpointIdGet(endpointId, errorsThreshold, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['SIPEndpointApi.apiV1SipEndpointStatusEndpointIdGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.SIPEndpointApiFp = SIPEndpointApiFp;
/**
 * SIPEndpointApi - factory interface
 * @export
 */
const SIPEndpointApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.SIPEndpointApiFp)(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointByNameGet(name, options) {
            return localVarFp.apiV1SipEndpointByNameGet(name, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} alias
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointCapacityByAliasGet(alias, options) {
            return localVarFp.apiV1SipEndpointCapacityByAliasGet(alias, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} endpointId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointCapacityEndpointIdGet(endpointId, options) {
            return localVarFp.apiV1SipEndpointCapacityEndpointIdGet(endpointId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointGet(skip, take, options) {
            return localVarFp.apiV1SipEndpointGet(skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdDelete(id, options) {
            return localVarFp.apiV1SipEndpointIdDelete(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdGet(id, options) {
            return localVarFp.apiV1SipEndpointIdGet(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options) {
            return localVarFp.apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointPost(sIPEndpointAddDTO, options) {
            return localVarFp.apiV1SipEndpointPost(sIPEndpointAddDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} endpointId
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointReportsEndpointIdGet(endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options) {
            return localVarFp.apiV1SipEndpointReportsEndpointIdGet(endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointReportsGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options) {
            return localVarFp.apiV1SipEndpointReportsGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} [fromCreatedTime]
         * @param {string} [toCreatedTime]
         * @param {number} [fromStatus]
         * @param {number} [toStatus]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointReportsStatusGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, options) {
            return localVarFp.apiV1SipEndpointReportsStatusGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} alias
         * @param {number} [errorsThreshold]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointStatusByAliasGet(alias, errorsThreshold, options) {
            return localVarFp.apiV1SipEndpointStatusByAliasGet(alias, errorsThreshold, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} endpointId
         * @param {number} [errorsThreshold]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointStatusEndpointIdGet(endpointId, errorsThreshold, options) {
            return localVarFp.apiV1SipEndpointStatusEndpointIdGet(endpointId, errorsThreshold, options).then((request) => request(axios, basePath));
        },
    };
};
exports.SIPEndpointApiFactory = SIPEndpointApiFactory;
/**
 * SIPEndpointApi - object-oriented interface
 * @export
 * @class SIPEndpointApi
 * @extends {BaseAPI}
 */
class SIPEndpointApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} name
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointByNameGet(name, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointByNameGet(name, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} alias
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointCapacityByAliasGet(alias, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointCapacityByAliasGet(alias, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} endpointId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointCapacityEndpointIdGet(endpointId, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointCapacityEndpointIdGet(endpointId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointGet(skip, take, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointGet(skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointIdDelete(id, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointIdDelete(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointIdGet(id, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointIdGet(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointPost(sIPEndpointAddDTO, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointPost(sIPEndpointAddDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} endpointId
     * @param {string} [fromCreatedTime]
     * @param {string} [toCreatedTime]
     * @param {number} [fromStatus]
     * @param {number} [toStatus]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointReportsEndpointIdGet(endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointReportsEndpointIdGet(endpointId, fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} [fromCreatedTime]
     * @param {string} [toCreatedTime]
     * @param {number} [fromStatus]
     * @param {number} [toStatus]
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointReportsGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointReportsGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} [fromCreatedTime]
     * @param {string} [toCreatedTime]
     * @param {number} [fromStatus]
     * @param {number} [toStatus]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointReportsStatusGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointReportsStatusGet(fromCreatedTime, toCreatedTime, fromStatus, toStatus, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} alias
     * @param {number} [errorsThreshold]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointStatusByAliasGet(alias, errorsThreshold, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointStatusByAliasGet(alias, errorsThreshold, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} endpointId
     * @param {number} [errorsThreshold]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointStatusEndpointIdGet(endpointId, errorsThreshold, options) {
        return (0, exports.SIPEndpointApiFp)(this.configuration).apiV1SipEndpointStatusEndpointIdGet(endpointId, errorsThreshold, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SIPEndpointApi = SIPEndpointApi;
