// @dasha.ai/sdk@0.11.2/dist/rest-api/generated/profiler/api/profiler-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Profiler API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilerApi = exports.ProfilerApiFactory = exports.ProfilerApiFp = exports.ProfilerApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ProfilerApi - axios parameter creator
 * @export
 */
const ProfilerApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerDownloadGet: async (start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options = {}) => {
            const localVarPath = `/api/v1/profiler/download`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (start !== undefined) {
                localVarQueryParameter['Start'] = (start instanceof Date) ?
                    start.toISOString() :
                    start;
            }
            if (end !== undefined) {
                localVarQueryParameter['End'] = (end instanceof Date) ?
                    end.toISOString() :
                    end;
            }
            if (applicationId !== undefined) {
                localVarQueryParameter['ApplicationId'] = applicationId;
            }
            if (applicationName !== undefined) {
                localVarQueryParameter['ApplicationName'] = applicationName;
            }
            if (createdAfter !== undefined) {
                localVarQueryParameter['CreatedAfter'] = (createdAfter instanceof Date) ?
                    createdAfter.toISOString() :
                    createdAfter;
            }
            if (clustersNumber !== undefined) {
                localVarQueryParameter['ClustersNumber'] = clustersNumber;
            }
            if (clusterAlgorithm !== undefined) {
                localVarQueryParameter['ClusterAlgorithm'] = clusterAlgorithm;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerDownloadSessionLogTableGet: async (start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options = {}) => {
            const localVarPath = `/api/v1/profiler/download/session-log-table`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (start !== undefined) {
                localVarQueryParameter['Start'] = (start instanceof Date) ?
                    start.toISOString() :
                    start;
            }
            if (end !== undefined) {
                localVarQueryParameter['End'] = (end instanceof Date) ?
                    end.toISOString() :
                    end;
            }
            if (applicationId !== undefined) {
                localVarQueryParameter['ApplicationId'] = applicationId;
            }
            if (applicationName !== undefined) {
                localVarQueryParameter['ApplicationName'] = applicationName;
            }
            if (createdAfter !== undefined) {
                localVarQueryParameter['CreatedAfter'] = (createdAfter instanceof Date) ?
                    createdAfter.toISOString() :
                    createdAfter;
            }
            if (clustersNumber !== undefined) {
                localVarQueryParameter['ClustersNumber'] = clustersNumber;
            }
            if (clusterAlgorithm !== undefined) {
                localVarQueryParameter['ClusterAlgorithm'] = clusterAlgorithm;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {ProfilerRequestDTO} profilerRequestDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerRequestPost: async (profilerRequestDTO, options = {}) => {
            // verify required parameter 'profilerRequestDTO' is not null or undefined
            (0, common_1.assertParamExists)('apiV1ProfilerRequestPost', 'profilerRequestDTO', profilerRequestDTO);
            const localVarPath = `/api/v1/profiler/request`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(profilerRequestDTO, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerStatusGet: async (start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options = {}) => {
            const localVarPath = `/api/v1/profiler/status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication oauth2 required
            // oauth required
            await (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            // authentication Bearer required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "Authorization", configuration);
            if (start !== undefined) {
                localVarQueryParameter['Start'] = (start instanceof Date) ?
                    start.toISOString() :
                    start;
            }
            if (end !== undefined) {
                localVarQueryParameter['End'] = (end instanceof Date) ?
                    end.toISOString() :
                    end;
            }
            if (applicationId !== undefined) {
                localVarQueryParameter['ApplicationId'] = applicationId;
            }
            if (applicationName !== undefined) {
                localVarQueryParameter['ApplicationName'] = applicationName;
            }
            if (createdAfter !== undefined) {
                localVarQueryParameter['CreatedAfter'] = (createdAfter instanceof Date) ?
                    createdAfter.toISOString() :
                    createdAfter;
            }
            if (clustersNumber !== undefined) {
                localVarQueryParameter['ClustersNumber'] = clustersNumber;
            }
            if (clusterAlgorithm !== undefined) {
                localVarQueryParameter['ClusterAlgorithm'] = clusterAlgorithm;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ProfilerApiAxiosParamCreator = ProfilerApiAxiosParamCreator;
/**
 * ProfilerApi - functional programming interface
 * @export
 */
const ProfilerApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ProfilerApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ProfilerDownloadGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ProfilerDownloadGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ProfilerApi.apiV1ProfilerDownloadGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ProfilerDownloadSessionLogTableGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ProfilerDownloadSessionLogTableGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ProfilerApi.apiV1ProfilerDownloadSessionLogTableGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {ProfilerRequestDTO} profilerRequestDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ProfilerRequestPost(profilerRequestDTO, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ProfilerRequestPost(profilerRequestDTO, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ProfilerApi.apiV1ProfilerRequestPost']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1ProfilerStatusGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1ProfilerStatusGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['ProfilerApi.apiV1ProfilerStatusGet']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.ProfilerApiFp = ProfilerApiFp;
/**
 * ProfilerApi - factory interface
 * @export
 */
const ProfilerApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ProfilerApiFp)(configuration);
    return {
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerDownloadGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
            return localVarFp.apiV1ProfilerDownloadGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerDownloadSessionLogTableGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
            return localVarFp.apiV1ProfilerDownloadSessionLogTableGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {ProfilerRequestDTO} profilerRequestDTO
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerRequestPost(profilerRequestDTO, options) {
            return localVarFp.apiV1ProfilerRequestPost(profilerRequestDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} [start]
         * @param {string} [end]
         * @param {string} [applicationId]
         * @param {string} [applicationName]
         * @param {string} [createdAfter]
         * @param {number} [clustersNumber]
         * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ProfilerStatusGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
            return localVarFp.apiV1ProfilerStatusGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ProfilerApiFactory = ProfilerApiFactory;
/**
 * ProfilerApi - object-oriented interface
 * @export
 * @class ProfilerApi
 * @extends {BaseAPI}
 */
class ProfilerApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} [start]
     * @param {string} [end]
     * @param {string} [applicationId]
     * @param {string} [applicationName]
     * @param {string} [createdAfter]
     * @param {number} [clustersNumber]
     * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProfilerApi
     */
    apiV1ProfilerDownloadGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
        return (0, exports.ProfilerApiFp)(this.configuration).apiV1ProfilerDownloadGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} [start]
     * @param {string} [end]
     * @param {string} [applicationId]
     * @param {string} [applicationName]
     * @param {string} [createdAfter]
     * @param {number} [clustersNumber]
     * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProfilerApi
     */
    apiV1ProfilerDownloadSessionLogTableGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
        return (0, exports.ProfilerApiFp)(this.configuration).apiV1ProfilerDownloadSessionLogTableGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {ProfilerRequestDTO} profilerRequestDTO
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProfilerApi
     */
    apiV1ProfilerRequestPost(profilerRequestDTO, options) {
        return (0, exports.ProfilerApiFp)(this.configuration).apiV1ProfilerRequestPost(profilerRequestDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} [start]
     * @param {string} [end]
     * @param {string} [applicationId]
     * @param {string} [applicationName]
     * @param {string} [createdAfter]
     * @param {number} [clustersNumber]
     * @param {ClusterAlgorithmEnum} [clusterAlgorithm]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProfilerApi
     */
    apiV1ProfilerStatusGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options) {
        return (0, exports.ProfilerApiFp)(this.configuration).apiV1ProfilerStatusGet(start, end, applicationId, applicationName, createdAfter, clustersNumber, clusterAlgorithm, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ProfilerApi = ProfilerApi;
