// @dasha.ai/sdk@0.11.2/dist/grpc-api/node/base.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = exports.Stream = void 0;
const eventemitter3_1 = __importDefault(require("eventemitter3"));
const grpc = __importStar(require("@grpc/grpc-js"));
const log_1 = require("../../log");
const errors = __importStar(require("../../errors"));
class Stream extends eventemitter3_1.default {
    constructor(init, logger) {
        super();
        this.isClosed = false;
        this.isEnd = false;
        this.isStatusError = false;
        this.isError = false;
        this.isCanceled = false;
        this.logger = logger;
        this._grpcStream = init(this);
        this._grpcStream.on("error", (error) => {
            this.logger.warn(`Stream unhealthy. Got message '${error.message}' with name '${error.name}'`, { message: error.message, name: error.name });
            this.isError = true;
        });
        this._grpcStream.on("close", () => {
            this.logger.debug("Stream close.");
            this.isClosed = true;
        });
        this._grpcStream.on("end", () => {
            this.logger.debug("Stream end.");
            this.isEnd = true;
        });
        this._grpcStream.on("status", (status) => {
            if (status.code !== grpc.status.OK) {
                this.logger.warn(`Stream unhealthy. Got '${status}' status.`, { status: status });
                this.isStatusError = true;
            }
        });
    }
    cancel() {
        this.removeAllListeners();
        this._grpcStream.removeAllListeners();
        this._grpcStream.on("error", (error) => void 0);
        // workaround for https://github.com/grpc/grpc-node/issues/1652
        setTimeout(() => {
            this._grpcStream.cancel();
        }, 0);
    }
    healthCheck() {
        const streamHealth = !this._grpcStream.destroyed && !this._grpcStream.readableEnded && !this._grpcStream.isPaused();
        const customHealth = !this.isCanceled && !this.isEnd && !this.isClosed && !this.isError && !this.isStatusError;
        if (streamHealth && customHealth) {
            this.logger.silly("Stream healthy.");
            return true;
        }
        const healthObject = {
            isCanceled: this.isCanceled,
            isEnd: this.isEnd,
            isClosed: this.isClosed,
            isError: this.isError,
            isStatusError: this.isStatusError,
            destroyed: this._grpcStream.destroyed,
            readableEnded: this._grpcStream.readableEnded,
            isPaused: this._grpcStream.isPaused()
        };
        this.logger.warn(`Stream unhealthy. '${JSON.stringify(healthObject)}'`);
        return false;
    }
    enrichLoggerMeta(meta) {
        this.logger = this.logger.child({ ...this.logger.defaultMeta, ...meta });
    }
}
exports.Stream = Stream;
class Client {
    constructor(Client, account, log) {
        this.log = log;
        const { server, apiKey, insecure } = account;
        const metadata = new grpc.Metadata();
        metadata.add("Authorization", `Bearer ${apiKey}`);
        if (account.customerId !== undefined) {
            metadata.add("X-CustomerId", account.customerId);
        }
        if (insecure) {
            this.grpcClient = new Client(server, grpc.credentials.createInsecure(), {
                callInvocationTransformer(callProperties) {
                    callProperties.metadata.merge(metadata);
                    return callProperties;
                },
                "grpc.keepalive_time_ms": 10000,
            });
        }
        else {
            this.grpcClient = new Client(server, grpc.credentials
                .createSsl()
                .compose(grpc.credentials.createFromMetadataGenerator((_, cb) => cb(null, metadata))), {
                "grpc.keepalive_time_ms": 10000,
                'grpc.service_config': JSON.stringify({ loadBalancingConfig: [{ round_robin: {} }], })
            });
        }
    }
    close() {
        this.grpcClient.close();
    }
    enrichLoggerMeta(meta) {
        this.log = this.log.child({ ...this.log.defaultMeta, ...meta });
    }
    healthCheck() {
        const channelState = this.grpcClient.getChannel().getConnectivityState(false);
        if (channelState >= grpc.connectivityState.TRANSIENT_FAILURE) {
            this.log.warn(`Channel unhealthy. '${JSON.stringify({ state: channelState })}'`, { state: channelState });
            return false;
        }
        this.log.silly(`Channel healthy. '${JSON.stringify({ state: channelState })}'`, { state: channelState });
        return true;
    }
    wrapGrpcError(error) {
        this.log.debug(error);
        switch (error.code) {
            case grpc.status.INVALID_ARGUMENT:
                return (0, log_1.logAndReturn)(this.log, new errors.ArgumentError(error.message));
            case grpc.status.PERMISSION_DENIED:
                return (0, log_1.logAndReturn)(this.log, new errors.PermissionError(error.message));
            case grpc.status.INTERNAL:
                if (error.message.includes("Received RST_STREAM with code 2") ||
                    error.message.includes("Received RST_STREAM with code 0") ||
                    error.message.includes("Failed to start HTTP/2 stream with error")) {
                    return (0, log_1.logAndReturn)(this.log, new errors.ConnectionError(error.message, error, true));
                }
                return (0, log_1.logAndReturn)(this.log, new errors.InternalError(error.message, error));
            case grpc.status.CANCELLED:
            case grpc.status.DEADLINE_EXCEEDED:
            case grpc.status.RESOURCE_EXHAUSTED:
            case grpc.status.ABORTED:
            case grpc.status.DATA_LOSS:
            case grpc.status.UNKNOWN:
            case grpc.status.UNAVAILABLE:
                return (0, log_1.logAndReturn)(this.log, new errors.ConnectionError(error.message, error, true));
            case grpc.status.UNAUTHENTICATED:
                return (0, log_1.logAndReturn)(this.log, new errors.AuthenticationError());
            default:
                return (0, log_1.logAndReturn)(this.log, new errors.InternalError(error.message, error));
        }
    }
}
exports.Client = Client;
