// @dasha.ai/sdk@0.11.2/dist/grpc-api/node/multichannel-text-protocol.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiChannelTextProtocolClient = void 0;
const base = __importStar(require("./base"));
const sdk_grpc_pb = __importStar(require("./generated/sdk_grpc_pb"));
const sdk_pb = __importStar(require("./generated/sdk_pb"));
class MultiChannelTextProtocolClient extends base.Client {
    constructor(account, log) {
        super(sdk_grpc_pb.TextProtocolClient, account, log.child({ ...log.defaultMeta, service: "multi-chat" }));
    }
    sendOpen(request) {
        return new Promise((resolve, reject) => {
            const { jobId, channelId, sequenceId } = request;
            const grpcRequest = new sdk_pb.TextProtocolMessage();
            grpcRequest.setJobId(jobId);
            grpcRequest.setOpened(new sdk_pb.OpenedSessionMessage().setChannelid(channelId).setSequenceId(sequenceId));
            this.grpcClient.sendMessage(grpcRequest, (grpcError) => {
                if (grpcError) {
                    reject(this.wrapGrpcError(grpcError));
                    return;
                }
                resolve();
            });
        });
    }
    sendFailed(request) {
        return new Promise((resolve, reject) => {
            const { jobId, channelId, sequenceId } = request;
            const grpcRequest = new sdk_pb.TextProtocolMessage();
            grpcRequest.setJobId(jobId);
            grpcRequest.setFailed(new sdk_pb.FailedSessionMessage().setChannelid(channelId).setSequenceId(sequenceId));
            this.grpcClient.sendMessage(grpcRequest, (grpcError) => {
                if (grpcError) {
                    reject(this.wrapGrpcError(grpcError));
                    return;
                }
                resolve();
            });
        });
    }
    sendText(request) {
        return new Promise((resolve, reject) => {
            const { jobId, text, channelId } = request;
            const grpcRequest = new sdk_pb.TextProtocolMessage();
            grpcRequest.setJobId(jobId);
            grpcRequest.setText(new sdk_pb.TextMessage().setText(text).setChannelid(channelId));
            this.grpcClient.sendMessage(grpcRequest, (grpcError) => {
                if (grpcError) {
                    reject(this.wrapGrpcError(grpcError));
                    return;
                }
                resolve();
            });
        });
    }
    sendClose(request) {
        return new Promise((resolve, reject) => {
            const { jobId, channelId } = request;
            const grpcRequest = new sdk_pb.TextProtocolMessage();
            grpcRequest.setJobId(jobId);
            grpcRequest.setClosed(new sdk_pb.ClosedSessionMessage().setChannelid(channelId));
            this.grpcClient.sendMessage(grpcRequest, (grpcError) => {
                if (grpcError) {
                    reject(this.wrapGrpcError(grpcError));
                    return;
                }
                resolve();
            });
        });
    }
    messageStream(request) {
        return new base.Stream((stream) => {
            const grpcRequest = new sdk_pb.TextProtocolStreamInit();
            grpcRequest.setFeatures(new sdk_pb.TextProtocolFeatures().setMultichannel(true));
            grpcRequest.setJobId(request.jobId);
            const grpcStream = this.grpcClient.messageStream(grpcRequest);
            grpcStream.on("error", (grpcError) => {
                stream.emit("error", this.wrapGrpcError(grpcError));
                stream.cancel();
            });
            grpcStream.on("end", () => {
                grpcStream.removeAllListeners();
                stream.removeAllListeners();
            });
            grpcStream.on("data", (grpcMessage) => {
                const { jobId, text, closed, open, close, inWait } = grpcMessage.toObject();
                if (text !== undefined) {
                    stream.emit("text", jobId, text.text, text.channelidsList);
                }
                if (open !== undefined) {
                    stream.emit("open", jobId, open.endpoint, open.channelid, open.sequenceId);
                }
                if (close !== undefined) {
                    stream.emit("close", jobId, close.channelid);
                }
                if (closed !== undefined) {
                    stream.emit("closed", jobId, closed.channelid);
                }
                if (inWait !== undefined) {
                    stream.emit("wait", jobId, inWait.channelid);
                }
            });
            return grpcStream;
        }, this.log.child({
            ...this.log.defaultMeta,
            streamName: "message"
        }));
    }
}
exports.MultiChannelTextProtocolClient = MultiChannelTextProtocolClient;
